### PR TITLE
refactor: Optimize GoogleScholarTool

### PR DESCRIPTION
**Pull request by koch3092**
## Description

`GoogleScholarTool.get_author_publications()` & `GoogleScholarTool.get_publication_by_title()` support passing in author

## Motivation and Context

Related #1121 

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/google_scholar_toolkit.py [hunk 1]
```diff
@@ -77,21 +77,25 @@ def get_author_detailed_info(
         return author
 
     def get_author_publications(
-        self,
+        self, author: Optional[dict] = None
```
Comment (Wendong-Fan) [inline camel/toolkits/google_scholar_toolkit.py:80]: how will user use this? they need manually set this dict?

Comment (koch3092) [reply camel/toolkits/google_scholar_toolkit.py:80]: Author is is initialized through `get_author_detailed_info()`. After saving the author's information, the user can continue to the next step to obtain other data.

Comment (Wendong-Fan) [reply camel/toolkits/google_scholar_toolkit.py:80]: I didn't get see benefit of the change

if user want to get publications from 2 authors, with current code:
```
author1_pub = GoogleScholarToolkit(author_identifier="id1").get_author_publications()

author2_pub = GoogleScholarToolkit(author_identifier="id2").get_author_publications()
``

with your code:
```
author1_pub = GoogleScholarToolkit(author_identifier="id1").get_author_publications()

author2_info = GoogleScholarToolkit(author_identifier="id2").get_author_detailed_info()
author2_pub = GoogleScholarToolkit(author_identifier="id2").get_author_publications(author2_info)
```

Comment (koch3092) [reply camel/toolkits/google_scholar_toolkit.py:80]: Here is a possible scenario:
1. First get the author information, such as name, homepage, email_domain, and save it.
2. Then the next step is to determine whether a certain title is the user's publication.

I can only implement it like the following code:
```
author_info = GoogleScholarToolkit(author_identifier="id1").get_author_detailed_info()
save(author_info["name"], author_info["homepage"], author_info["email_domain"])
publications = author['publications']

for publication in publications:
    if publication['bib']['title'] == publication_title:
        return self.scholarly.fill(publication)
```

Or after saving the information, I want to get the title list:
```
author_info = GoogleScholarToolkit(author_identifier="id1").get_author_detailed_info()
save(author_info["name"], author_info["homepage"], author_info["email_domain"])
publications = author['publications']

publication_titles = [
    pub['bib']['title'] for pub in author['publications']
]
```

However, existing methods cannot be called.

Comment (Wendong-Fan) [reply camel/toolkits/google_scholar_toolkit.py:80]: Screenshot 2024-10-25 at 21 48 28

what's this `publication_title`?

Comment (Wendong-Fan) [reply camel/toolkits/google_scholar_toolkit.py:80]: and why we need  to determine whether a certain title is the user's publication

Comment (koch3092) [reply camel/toolkits/google_scholar_toolkit.py:80]: > Screenshot 2024-10-25 at 21 48 28
> 
> what's this `publication_title`?

It's just a hypothesis that if there is a need, how do you do it?

**Comment by Wendong-Fan** [comment]
close as finished in https://github.com/camel-ai/camel/pull/1124

**Pull request closed without merging**